modules/caddyhttp/replacer.go

```diff
@@ -142,8 +142,16 @@ func addHTTPVarsToReplacer(repl *caddy.Replacer, req *http.Request, w http.Respo
 				}
 				return port, true
 			case "http.request.remote":
+				if req.TLS != nil && !req.TLS.HandshakeComplete {
+					// without a complete handshake (QUIC "early data") we can't trust the remote IP address to not be spoofed
+					return nil, true
+				}
 				return req.RemoteAddr, true
 			case "http.request.remote.host":
+				if req.TLS != nil && !req.TLS.HandshakeComplete {
+					// without a complete handshake (QUIC "early data") we can't trust the remote IP address to not be spoofed
+					return nil, true
+				}
```

Comment:
Perhaps do the same for the `http.request.remote.port` below

Reply:
I thought about it, but I've never heard of a port being spoofed (and why?)... not sure how useful it is to reject that too. I'm open to it though if that's the right thing to do.

Reply:
You're right. I was just thinking about it from consistency perspective. I also can't think of a security issue coming with it.